core/src/main/java/org/apache/calcite/sql/dialect/SnowflakeSqlDialect.java

```diff
@@ -120,10 +120,14 @@ public SnowflakeSqlDialect(Context context) {
   }
 
   @Override public SqlOperator getTargetFunc(RexCall call) {
-    switch (call.type.getSqlTypeName()) {
-    case DATE:
-    case TIMESTAMP:
-      return getTargetFunctionForDateOperations(call);
+    switch (call.getOperator().getKind()) {
```

Comment:
Please remove all such **PLUS** and **MINUS** checks, as we already have a case check for the same in the calling method.

Reply:
done. removed.